torch_geometric/nn/norm/batch_norm.py

```diff
@@ -30,18 +30,41 @@ class BatchNorm(torch.nn.Module):
             :obj:`False`, this module does not track such statistics and always
             uses batch statistics in both training and eval modes.
             (default: :obj:`True`)
+        allow_no_batch (bool, optional): If set to :obj:`True`, batches with
+            only a single element will work as though in training mode. That is
+            the running mean and variance will be used.
+            (default: :obj:`False`)
     """
-    def __init__(self, in_channels, eps=1e-5, momentum=0.1, affine=True,
-                 track_running_stats=True):
+    def __init__(self, in_channels: int, eps: float = 1e-5,
+                 momentum: float = 0.1, affine: bool = True,
+                 track_running_stats: bool = True,
+                 allow_no_batch: bool = False):
         super().__init__()
         self.module = torch.nn.BatchNorm1d(in_channels, eps, momentum, affine,
                                            track_running_stats)
+        self.in_channels = in_channels
+        self.allow_no_batch = allow_no_batch
 
     def reset_parameters(self):
         self.module.reset_parameters()
 
     def forward(self, x: Tensor) -> Tensor:
         """"""
+        if self.allow_no_batch and x.size(0) <= 1:
+            training = self.module.training
+            running_mean = self.module.running_mean
+            running_var = self.module.running_var
+            if running_mean is None or running_var is None:
```

Comment:
I am still not sure about this while looking at https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/batchnorm.py#L53. I think `running_mean` is only `None` in case of `track_running_stats=False`, in which case we should also error out.

Reply:
Oh hmm, yes, you're right, the problem only happens if `track_running_stats` is `False`.  

In our current implementation, if `track_running_stats=False` we do not raise an exception in training mode. What would be the logic for us to do so in this specific case?

Reply:
The tests in `test/nn/norm/test_batch_norm.py` all pass with this implementation, removing this will throw an error in the single combination
- track_running_stats=False
- batch_size=1

Reply:
Yes, it is impossible to support `track_running_stats=False` with `batch_size=1`. The current implementation does simply not do any normalization which is probably not desired. I would simply error out in this case TBH.

Reply:
Hmm yeah I guess that makes the most sense to me. Its a bit ambiguous but I haven't got a concrete use case for supporting this either so have updated based on your suggestion.